pom.xml

```diff
@@ -298,4 +298,37 @@ tools.jackson.core.*;version=${project.version}
       <scope>test</scope>
     </dependency>
   </dependencies>
+
+  <profiles>
+    <profile>
+      <id>openrewrite</id>
+      <!-- `mvn -P openrewrite org.openrewrite.maven:rewrite-maven-plugin:run` -->
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.openrewrite.maven</groupId>
+            <artifactId>rewrite-maven-plugin</artifactId>
+            <version>5.41.0</version>
```

Comment:
This can't be used on the 2.x branches because it would cause a huge diff

Reply:
I can try to put the info from the profile directly into the command in the workflow, I'll just have to figure out how to specify the version number there.
The advantage of having the profile is in the pom is that you can reuse it to apply the changes locally. On the other hand if all the information is in the workflow then the workflow does not depend on the pom and you do not run the risk that changes to the pom inadvertently change the workflow behavior so this might be safer.

Reply:
We have the imports ordered on master so we can afford to support this and have full reordering - it is just not a good idea to have this on the 2.x branches.

I still have my objections to using any build tooling that doesn't have multitudes of users and oodles of history to establish trust.

My preference is not merge any of this. Low cost to benefit ratio to me.

If we insist on merging something like this - I would prefer to look at established tools to do this.

I code mainly in Scala and use Scalafmt a lot. But if someone came to me with a custom project and asked me to replace Scalafmt, I would need a load of convincing.

Reply:
I see your point. Openrewrite is still new. It is the best tool I know for automatic refactoring and by extension it can also reformat code to a custom style. 
_(tangent: I believe that it is a time sink to define a [custom code style](https://github.com/FasterXML/jackson/blob/master/contribution/jackson-coding-style-guide.md#jackson-coding-style-guide) without also having tools to enforce it. Thus my main suggestion remains to [use an established code style](https://github.com/FasterXML/jackson-databind/pull/4448#discussion_r1544730223) which can be easily enforced. But @cowtowncoder is [not fan](https://github.com/FasterXML/jackson/pull/201#issuecomment-1848851178) of comprehensive auto-formatters and the google formatter in particular and them having fun maintaining the project is ultimately more important to its success than adherence to a code style.)_
I tried to see if I can use Openrewrite to enforce key aspects of the code style on PRs and thereby relieve the reviewer. I think such a workflow is really useful to fix common sonarqube findings on PRs in an industrial setting. However, I'm not sure if there are enough PRs on this project to justify adding that complexity to the project.

Reply:
Again trying not to cast stones - but your PR includes call to a lib that is under your control. 
io.github.timo-a:rewrite-recipe-starter - this is a project with 0 stars, 0 forks